projects/core/src/organization/facade/budget.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { combineLatest } from 'rxjs';
+import { debounceTime, filter, map, take, tap } from 'rxjs/operators';
+
+import { getProcessStateFactory } from '../../process/store/selectors/process.selectors';
+import { StateWithProcess } from '../../process/store/process-state';
+import { LoaderState } from '../../state/utils/loader/loader-state';
+import { AuthService } from '../../auth/facade/auth.service';
+import { Budget } from '../../model/budget.model';
+import {
+  LOAD_BUDGETS_PROCESS_ID,
+  StateWithOrganization,
+} from '../store/organization-state';
+import { BudgetActions } from '../store/actions/index';
+import {
+  getBudgetsState,
+  getBudgetValueState,
+  getBudgetState,
+} from '../store/selectors/budget.selector';
+
+@Injectable()
+export class BudgetService {
+  private user$ = this.authService.getOccUserId();
+
+  constructor(
+    protected store: Store<StateWithOrganization | StateWithProcess<void>>,
+    protected authService: AuthService
+  ) {}
+
+  loadBudget(budgetCode: string) {
+    this.user$
+      .pipe(take(1))
+      .subscribe(uid =>
+        this.store.dispatch(new BudgetActions.LoadBudget({ uid, budgetCode }))
+      );
+  }
+
+  loadBudgets() {
+    this.user$
+      .pipe(take(1))
+      .subscribe(uid =>
+        this.store.dispatch(new BudgetActions.LoadBudgets(uid))
+      );
+  }
+
+  getBudgetsProcess() {
+    return this.store.select(getProcessStateFactory(LOAD_BUDGETS_PROCESS_ID));
+  }
+
+  getBudgetValue(budgetCode: string) {
+    return this.store.select(getBudgetValueState(budgetCode));
+  }
+
+  getBudgetState(budgetCode: string) {
+    return this.store.select(getBudgetState(budgetCode));
+  }
+
+  getBudgets() {
+    return this.store.select(getBudgetsState);
+  }
+
+  loadAndGetBudgets() {
```

Comment:
The API of this service looks a bit massive. Let's reduce the methods for imperative loading and the method to get process loading.
It would be nice to have 4 simple methods in BudgetService 
- get
- getList
- update
- create
(the rest could be protected).

The get & getList should do the logic of current loadAndGetBudgets etc.
Then in the components you simply don't care about loading stuff, you are interested in emitted/not-emitted values.